framework/src/positions/FunctorExtremaPositions.C

```diff
@@ -0,0 +1,173 @@
+  else if (getParam<ExecFlagEnum>("execute_on").contains(EXEC_NONE))
+    paramError("functor", "NONE execution flag not supported for a function.");
+}
+
+void
+FunctorExtremaPositions::initialize()
+{
+  clearPositions();
+  _positions.resize(_n_extrema);
+  _positions_values.resize(_n_extrema);
+
+  std::vector<Real> extrema;
+  std::vector<Point> extrema_locs;
+  if (_type == ExtremeType::MAX || _type == ExtremeType::MAX_ABS)
+    extrema.resize(_n_extrema, -std::numeric_limits<Real>::max());
+  else
+    extrema.resize(_n_extrema, std::numeric_limits<Real>::max());
+  extrema_locs.resize(_n_extrema);
+
+  const auto time_arg = determineState();
+
+  // Loop over the local mesh, keep track of all extrema
+  for (const auto & elem :
+       _fe_problem.mesh().getMesh().active_local_subdomain_set_element_ptr_range(blockIDs()))
+  {
+    const Moose::ElemArg elem_arg = {elem, false};
+    auto value = _functor(elem_arg, time_arg);
+    for (const auto i : make_range(_n_extrema))
+    {
+      bool extrema_found = false;
+      switch (_type)
+      {
+        case ExtremeType::MAX_ABS:
+          value = std::abs(value);
+          // fallthrough
+        case ExtremeType::MAX:
+        {
+          if (value > extrema[i])
+            extrema_found = true;
+          break;
+        }
+        case ExtremeType::MIN:
+        {
+          if (value < extrema[i])
+            extrema_found = true;
+          break;
+        }
+      }
+      if (extrema_found)
+      {
+        // Move the extrema down the list of extrema
+        Real temp = value;
+        Point temp_loc = elem->true_centroid();
+        for (const auto j : make_range(i, _n_extrema))
+        {
+          const auto other_temp = extrema[j];
+          const auto other_temp_loc = extrema_locs[j];
+          extrema[j] = temp;
+          extrema_locs[j] = temp_loc;
+          temp = other_temp;
+          temp_loc = other_temp_loc;
+        }
+        // Found an extremum, filled the extrema vector, done
+        break;
+      }
+    }
+  }
+
+  // Synchronize across all ranks
+  unsigned int current_candidate = 0;
+  for (const auto i : make_range(_n_extrema))
+  {
+    unsigned int rank = 0;
+    // dont modify the extremum, it might be the global n-th extremum in a later call
+    auto copy = extrema[current_candidate];
+    if (_type == ExtremeType::MAX || _type == ExtremeType::MAX_ABS)
+      comm().maxloc(copy, rank);
+    else
+      comm().minloc(copy, rank);
+
+    RealVectorValue extreme_point(0., 0., 0.);
+    if (rank == processor_id())
+    {
+      extreme_point = extrema_locs[current_candidate];
+      // Our candidate for ith max got accepted, move on to offering the next extremal one
+      current_candidate++;
+    }
+
+    // Send the position
+    // The broadcast and scatter are for root 0 sending to all. The sum works just as well
+    comm().sum(extreme_point);
+    _positions[i] = extreme_point;
+    _positions_values[i] = copy;
+  }
+
```

Comment:
I would try to `mooseAssert(comm().verify(_positions))` and `mooseAssert(comm().verify(_positions_values))` (at least in debug mode, maybe not devel) just for my own sanity of being sure this is working. Almost all the logic looks good to me, but it's good to double check

Reply:
It's a cool idea. let s put the first one in the base class